WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -0,0 +1,184 @@
+#import "DisplayableImageHelper.h"
+
+static const NSInteger FeaturedImageMinimumWidth = 640;
+
+@implementation DisplayableImageHelper
+
+/**
+ Get the url path of the image to display for a post.
+
+ @param dict A dictionary representing a posts attachments from the REST API.
+ @return The url path for the featured image or nil
+ */
++ (NSString *)searchPostAttachmentsForImageToDisplay:(NSDictionary *)attachmentsDict
+{
+    NSArray *attachments = [[attachmentsDict dictionaryForKey:@"attachments"] allValues];
+    if ([attachments count] == 0) {
+        return nil;
+    }
+
+    NSString *imageToDisplay;
+
+    attachments = [self filteredAttachmentsArray:attachments];
+
+    NSDictionary *attachment = [attachments firstObject];
+    NSInteger width = [[attachment numberForKey:@"width"] integerValue];
+    if (width >= FeaturedImageMinimumWidth) {
+        imageToDisplay = [attachment stringForKey:@"URL"];
+    }
+
+    return imageToDisplay;
+}
+
++ (NSArray *)filteredAttachmentsArray:(NSArray *)attachments
+{
+    NSString *key = @"mime_type";
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K BEGINSWITH %@", key, @"image"];
+    attachments = [attachments filteredArrayUsingPredicate:predicate];
+
+    return [attachments sortedArrayUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+        NSDictionary *attachmentA = (NSDictionary *)obj1;
+        NSDictionary *attachmentB = (NSDictionary *)obj2;
+        NSString *key = @"width";
```

Comment:
I'm just commenting once on constants, but this applies to a few more lines in `NSDate+StringFormatting.m` too:

I know this is part of a larger PR so please forgive me if this is coming up but... is there a better place where we can unify these keys?  Some file such as `WPAttachment` and name it `WPAttachmentSizeKey`? (just a thought, it does not need to be precisely that)